controller_manager/include/controller_manager/controller_manager.h

```diff
@@ -178,7 +183,11 @@ class ControllerManager{
   std::vector<controller_interface::ControllerBase*> start_request_, stop_request_;
   std::list<hardware_interface::ControllerInfo> switch_start_list_, switch_stop_list_;
   bool please_switch_;
+  bool switch_started_;
   int switch_strictness_;
+  bool start_asap_;
+  double timeout_;
+  ros::Time init_switch_;
```

Comment:
Is it outlandish to merge this with `switch_started_`?
This value oculd be wrapped in a `boost::optional` and used as a boolean in boolean expressions, then as a time value when needed.

Reply:
I don't see what you mean. `switch_started_` is used to avoid calling `doSwitch(...)` repetitively. It's not really related to the rest of the variables.

Reply:
Fixed with the below refactor of the switch parameters?